### PR TITLE
Plugin implementation

### PR DESCRIPTION
**Pull request by vitlinda**
This PR introduces the main features of the plugin:

- gets the source folder where to get the html files
- gets the target folder where to put the markdown files
- generates markdown tables using the specified html tags

### CODE REVIEW

File: src/main/scala/dev/atedeg/UbiquitousScaladoc.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        name <- document >?> text(t1)
+        description <- document >?> text(t2)
+      } yield Concept(name, description)
+    }
+
+    def tableBuilder: Builder = new Builder() withAlignment Table.ALIGN_LEFT addRow (tableHeaders *)
+
+    def generateMarkdownFile(dirName: String, concepts: Iterator[Concept], targetDir: File): Unit = {
+      val table = addConceptsToTable(concepts)
+      val file: File = targetDir / s"$dirName$fileNameSuffix$fileNameExtension"
```
Comment (giacomocavalieri) [inline src/main/scala/dev/atedeg/UbiquitousScaladoc.scala:53]: I think you do not need the type annotation here

File: src/sbt-test/sbt-us/simple/build.sbt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+ThisBuild / ubiquitousScaladoc / usSourceHtmlDir := file("api")
+ThisBuild / ubiquitousScaladoc / usTargetMarkdownDir := file("customTarget")
+
+lazy val root = (project in file("."))
+  .settings(
+    Compile / doc / target := baseDirectory.value / "api/generated"
```
Comment (nicolasfara) [inline src/sbt-test/sbt-us/simple/build.sbt:6]: Use the DSL for directory separation `Compile / doc / target := baseDirectory.value / "api" / "generated"`

File: src/sbt-test/sbt-us/simple/project/plugins.sbt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+resolvers += "jitpack" at "https://jitpack.io"
+sys.props.get("plugin.version") match {
+  case Some(v) => addSbtPlugin("dev.atedeg" % "sbt-ubiquitous-scaladoc" % v)
+  case _ => sys.error("No version specified")
+}
```
Comment (nicolasfara) [inline src/sbt-test/sbt-us/simple/project/plugins.sbt:5]: Missing trailing line

File: src/sbt-test/sbt-us/simple/src/main/scala/dev.atedeg.example/Example.scala [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Example doc.
+ */
+case class Example()
```
Comment (nicolasfara) [inline src/sbt-test/sbt-us/simple/src/main/scala/dev.atedeg.example/Example.scala:4]: Missing trailing line

File: src/main/scala/dev/atedeg/UbiquitousScaladocKeys.scala [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package dev.atedeg
+
+import sbt.*
+
+trait UbiquitousScaladocKeys {
+
+  val ubiquitousScaladoc: TaskKey[Unit] =
+    taskKey[Unit]("Task for creating a Ubiquitous Language markdown table out of a Scaladoc html file")
+
+  val usSourceHtmlDir: SettingKey[File] =
```
Comment (nicolasfara) [inline src/main/scala/dev/atedeg/UbiquitousScaladocKeys.scala:10]: Why the SettingKey's name start with `us`?
`sourceHtmlDir` is already taken inside SBT?

File: project/plugins.sbt [hunk 6]
```diff
@@ -1,2 +1,3 @@
 addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.6")
-addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.10.0")
\ No newline at end of file
+addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.10.0")
+addSbtPlugin("it.nicolasfarabegoli" % "sbt-conventional-commits" % "2.2.1")
```
Comment (nicolasfara) [inline project/plugins.sbt:3]: Missing trailing line

**Comment by nicolasfara** [comment]
I would suggest a bunch of things for manage the project (might be usefull also for other projects):
  1. enable renovate for this project
  2. configure semantic release
  3. Try to merge the initial PRs to `alpha` or `beta` branches to study the behaviour of pre-release process

**Review by giacomocavalieri — APPROVED** [review]

**Review by ndido98 — APPROVED** [review]

**Review by giacomocavalieri — APPROVED** [review]

**Review by nicolasfara — APPROVED** [review]

**Pull request merged**